### DISCUSSION TITLE
エラーによって起動するも使用できません

### DISCUSSION THREAD

**Discussion by yukkuri895** [tetsuya-ki/discord-live-notificationbot#23] [Q&A] [answered]
以前からニコ生の通知用として活用させていただいておりました。
先日サーバーを再起動して以降、MAC check failedエラーによって使用できなくなってしまいました。。
サーバーはRaspberry pi 4BでOSはRaspberryPi OS11.7です。以下に出力されたメッセージを貼りましたので、お時間があればよろしくお願いします。

> 2023-09-28 23:20:28,272@ discord.client [WARNING] __init__: PyNaCl is not installed, voice will NOT be supported
> 2023-09-28 23:20:28,273@ live-notification-bot [INFO] __init__: cogを読むぞ！
> 2023-09-28 23:20:28,454@ live-notification-bot [INFO] setup: LiveNotificationBotを読み込む！
> 2023-09-28 23:20:28,457@ discord.client [INFO] login: logging in using static token
>  * Serving Flask app "" (lazy loading)
>  * Environment: production
>    WARNING: This is a development server. Do not use it in a production deployment.
>    Use a production WSGI server instead.
>  * Debug mode: off
> 2023-09-28 23:20:29,090@ discord.gateway [INFO] identify: Shard ID None has sent the IDENTIFY payload.
> 2023-09-28 23:20:29,423@ discord.gateway [INFO] received_message: Shard ID None has connected to Gateway: ["gateway-prd-us-east1-b-tw1m",{"micros":131869,"calls":["id_created",{"micros":1606,"calls":[]},"session_lookup_time",{"micros":414,"calls":[]},"session_lookup_finished",{"micros":15,"calls":[]},"discord-sessions-prd-2-160",{"micros":128610,"calls":["start_session",{"micros":93466,"calls":["discord-api-5dcdf7bc48-lmknl",{"micros":86754,"calls":["get_user",{"micros":24979},"get_guilds",{"micros":10062},"send_scheduled_deletion_message",{"micros":13},"guild_join_requests",{"micros":328},"authorized_ip_coro",{"micros":14}]}]},"starting_guild_connect",{"micros":59,"calls":[]},"presence_started",{"micros":311,"calls":[]},"guilds_started",{"micros":113,"calls":[]},"guilds_connect",{"micros":2,"calls":[]},"presence_connect",{"micros":34602,"calls":[]},"connect_finished",{"micros":34609,"calls":[]},"build_ready",{"micros":20,"calls":[]},"clean_ready",{"micros":1,"calls":[]},"optimize_ready",{"micros":28,"calls":[]},"split_ready",{"micros":0,"calls":[]}]}]}] (Session ID: da832d0d0035e4687c1a20d59c393c4a).
> 2023-09-28 23:20:31,473@ discord_slash [INFO] sync_all_commands: Syncing commands...
> 2023-09-28 23:20:31,477@ live-notification-bot [INFO] on_ready: We have logged in as BotName#0000
> Ignoring exception in on_ready
> Traceback (most recent call last):
>   File "/home/pi/nico/discord-live-notificationbot/.venv/lib/python3.9/site-packages/discord/client.py", line 343, in _run_event
>     await coro(*args, **kwargs)
>   File "/home/pi/nico/discord-live-notificationbot/cogs/livenotificationcog.py", line 30, in on_ready
>     await self.liveNotification.prepare()  # dbを作成
>   File "/home/pi/nico/discord-live-notificationbot/cogs/modules/live_notification.py", line 110, in prepare
>     self.decode()
>   File "/home/pi/nico/discord-live-notificationbot/cogs/modules/live_notification.py", line 307, in decode
>     self.aes.decode()
>   File "/home/pi/nico/discord-live-notificationbot/cogs/modules/aes_angou.py", line 33, in decode
>     data = cipher.decrypt_and_verify(ciphertext, tag)
>   File "/home/pi/nico/discord-live-notificationbot/.venv/lib/python3.9/site-packages/Crypto/Cipher/_mode_eax.py", line 368, in decrypt_and_verify
>     self.verify(received_mac_tag)
>   File "/home/pi/nico/discord-live-notificationbot/.venv/lib/python3.9/site-packages/Crypto/Cipher/_mode_eax.py", line 309, in verify
>     raise ValueError("MAC check failed")
> ValueError: MAC check failed
> 2023-09-28 23:20:32,220@ discord_slash [INFO] sync_all_commands: Syncing permissions...
> 2023-09-28 23:20:32,221@ discord_slash [INFO] sync_all_commands: Completed syncing all commands!
> 2023-09-28 23:20:42,951@ live-notification-bot [INFO] live_notification_list: live-notificationを確認するぜ！
> 2023-09-28 23:20:42,952@ live-notification-bot [INFO] check_printer_is_running: Taskが停止していたので再起動します。
> Unhandled exception in internal background task 'printer'.
> Traceback (most recent call last):
>   File "/home/pi/nico/discord-live-notificationbot/.venv/lib/python3.9/site-packages/discord/ext/tasks/__init__.py", line 101, in _loop
>     await self.coro(*args, **kwargs)
>   File "/home/pi/nico/discord-live-notificationbot/cogs/livenotificationcog.py", line 57, in printer
>     for live in self.liveNotification.live_rows:
> TypeError: 'NoneType' object is not iterable

**Comment by tetsuya-ki** [comment]
うーん、、、暗号化したやつが正常に復号できなくて出るエラーっぽいですね。。。
適当に保存や暗号化などしているので、破壊されてしまったのかもしれないです。。。

**Reply by yukkuri895** [reply]
なるほど。。実行環境がローカルなので暗号化は解除しても問題はないと思われるので、何か方法はありますでしょうか...

**Reply by tetsuya-ki** [reply, marked as answer]
壊れた暗号化ファイルしかない状態の場合は、一旦ファイルを削除して作り直すしかないと思います(オプションによっては、Discordにそこそこ新しいバージョンのファイルが添付されてるかもしれません)。

また、以下オプションつけて起動すると暗号化処理が入らないので壊れにくいです。
KEEP_DECRYPTED_FILE=TRUE

お手数ですが、たまにバックアップ取るといいと思います。

**Reply by yukkuri895** [reply]
ありがとうございます！./cogs/modules/files/encrypted.binの削除で正常に動きました！
